great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1243,3 +1257,123 @@ def _create_validator(self, *, result_format: ResultFormatUnion) -> V1Validator:
+
+            >>> batch.compute_metrics([
+            ...     BatchRowCount(),
+            ...     ColumnMax(column="age")
+            ... ])
+            [BatchRowCountResult(id=..., value=1000), ColumnMaxResult(id=..., value=85)]
+
+        Notes:
+            Until this mypy bug is resolved, lists of Metrics are being incorrectly
+            inferred by the static type checker as list[Domain]. You can work around
+            this by adding an explicit annotation (e.g. metrics: list[Metric] = ...)
+            https://github.com/python/mypy/issues/18712
+        """
+        if isinstance(metrics, Metric):
+            metrics = [metrics]
+
+        execution_engine = self.data.execution_engine
+        execution_engine.batch_manager.load_batch_list(batch_list=[self])
+        metrics_calculator = MetricsCalculator(execution_engine=execution_engine)
+
+        requested_metric_names = [metric.name for metric in metrics]
+        metrics_calculator_result = metrics_calculator.compute_metrics(
+            metric_configurations=[metric.config for metric in metrics],
+            runtime_configuration=None,
+        )
+        return self.metrics_calculator_result_to_metric_result(
+            requested_metric_names=requested_metric_names,
+            metrics_calculator_result=metrics_calculator_result,
+        )
+
+    @classmethod
+    def metrics_calculator_result_to_metric_result(
+        cls,
+        requested_metric_names: list[str],
+        metrics_calculator_result: tuple[_MetricsDict, _AbortedMetricsInfoDict],
+    ) -> MetricResult | list[MetricResult]:
+        """Converts the result of metrics_calculator.compute_metrics() into
+        a MetricResult or list[MetricResult]. Only the metrics that were
+        requested in the call to metrics_calculator.compute_metrics()
+        will be returned (dependency metrics are excluded)."""
+        # add another level of nesting to these metric dicts
+        # since the key we care about is nested inside the current key
+        # enables us to iterate through these dicts only once
+        raw_metrics = {k.metric_name: (k, v) for k, v in metrics_calculator_result[0].items()}
+        raw_metrics_errors = {
+            k.metric_name: (k, v) for k, v in metrics_calculator_result[1].items()
+        }
+        metric_results = []
+        metric_types: list[type[Metric]] = cls.get_all_metric_classes()
+        for metric_name in requested_metric_names:
+            if metric_name in raw_metrics:
+                MetricResultType = cls.get_metric_result_type_from_metric_name(
+                    metric_name=metric_name,
+                    metric_types=metric_types,
+                )
+                metric_results.append(
+                    MetricResultType(
+                        id=raw_metrics[metric_name][0],
+                        value=raw_metrics[metric_name][1],
+                    )
+                )
+            if metric_name in raw_metrics_errors:
```

Comment:
Can a metric return both a result and an error? If not, I'd make this an `elif`.

Reply:
I'm pretty sure from reading [this try/except](https://github.com/great-expectations/great_expectations/blob/b0ca52840f6438f33259db447528cb2ebe78dc9c/great_expectations/validator/validation_graph.py#L275) the answer is no, it's one or the other. It's not the easiest to interpret given it all being in a while loop, and the successful metrics graph is resolved and updated in-place. I will update to `elif`.